### PR TITLE
feat(llm): accept /v1/messages InputFormat for chat/completions backends

### PR DESCRIPTION
**Pull request by apexlnc**
- implement Anthropic /v1/messages <-> OpenAI-style /v1/chat/completions translation across request, non-stream response, stream, and error paths
- extend role/content mapping in both directions, including tool calls, tool results, system/developer handling, and multimodal user content
- improve streaming translation sequencing and finish behavior (tool block lifecycle, done handling, and missing-finish edge cases)
- wire Messages InputFormat behavior consistently for OpenAI-compatible providers (OpenAI, Azure OpenAI, Vertex non-Anthropic, Gemini)
- fix Bedrock tool-result mapping for completions/responses translations to avoid forcing success status when source APIs do not carry explicit status
- add/expand translation tests for request/response/stream/error parity, roundtrip semantics, and edge-case regressions; refresh affected fixtures/snapshots
- bump async-openai to 0.32.4

### CODE REVIEW

**Comment by apexlnc** [comment]
Need this to go in first : https://github.com/agentgateway/agentgateway/pull/873 

LLM providers returning compressed SSE for responses that _we need to parse_ will break (for example claude code - > vertex gemini via `chat/completions`)

**Comment by apexlnc** [comment]
Probably still a bit too early, but opened this https://github.com/agentgateway/agentgateway/issues/910 in case we want to consider [`openresponses`](https://github.com/openresponses/openresponses) (OpenAI's `responses` schema) as the "default" IR

File: crates/agentgateway/src/llm/conversion/completions.rs [hunk 1]
```diff
@@ -251,55 +845,61 @@ pub mod from_messages {
...
-				Some(completions::Stop::StringArray(stop_sequences))
-			},
+			max_tokens: None,
+			stop,
 			tools: if tools.is_empty() { None } else { Some(tools) },
 			tool_choice,
-			parallel_tool_calls: None,
 			user: user_id,
-
 			vendor_extensions: completions::RequestVendorExtensions {
```
Comment (howardjohn) [inline crates/agentgateway/src/llm/conversion/completions.rs]: hmm this seems wrong. i think we really added this so you can go from completions --> something else, and set additional fields.

But using them from messages and sending them to openai is broken; openai will reject them.

File: crates/agentgateway/src/llm/conversion/completions.rs [hunk 2]
```diff
@@ -251,55 +845,61 @@ pub mod from_messages {
...
+			} else {
+				None
+			},
 			frequency_penalty: None,
+			presence_penalty: None,
+			seed: None,
+			// Fields not applicable from Anthropic Messages
+			store: None,
+			reasoning_effort,
+			metadata: metadata_json,
```
Comment (howardjohn) [inline crates/agentgateway/src/llm/conversion/completions.rs]: metadata requires `store` which we always have false

**Review by howardjohn — COMMENTED** [review]
@codex review

File: crates/agentgateway/tests/tests/llm.rs [hunk 3]
```diff
@@ -265,177 +483,160 @@ mod anthropic {
...
-	async fn embeddings() {
-		let Some(gw) = setup("azureOpenAI", "", "text-embedding-3-small").await else {
-			return;
-		};
-		send_embeddings(&gw, None).await;
-	}
+	send_completions_tests!("azureOpenAI", "", "gpt-4o-mini");
+	send_completions_tool_tests!("azureOpenAI", "", "gpt-4o-mini");
+	send_messages_tests!("azureOpenAI", "", "gpt-4o-mini");
+	send_messages_image_tests!(
```
Comment (howardjohn) [inline crates/agentgateway/tests/tests/llm.rs:610]: this one doesn't seem to work: 

```
ocessing failed: failed to parse response: missing field `type` at line 1 column 376","reason":"Internal","duration":"699ms","req.id":"64bd670187ea4
```

Comment (howardjohn) [reply crates/agentgateway/tests/tests/llm.rs:610]: ok actualyl this may just be a 429 lol. maybe we should be capturing error codes sooner before we parse the response... or maybe we are failing to parse the error I suppose. it looks like 

```
{"error":{"code":"RateLimitReached","message": "Your requests to gpt-4o-mini for gpt-4o-mini in East US have exceeded the token rate limit for your current OpenAI S0 pricing tier. This request was for ChatCompletions_Create under OpenAI Language Model Instance API. Please retry after 60 seconds. To increase your default rate limit, visit: https://aka.ms/oai/quotaincrease."}}
```

**Review by howardjohn — COMMENTED** [review]
with the tweaks I mentioned this seems to work with Claude code

File: crates/agentgateway/src/llm/conversion/completions.rs (hunk 1 above)
Comment (howardjohn) [reply crates/agentgateway/src/llm/conversion/completions.rs]: This is still a problem

**Comment by howardjohn** [comment]
`conversion::completions::from_messages::translate_stream` and some others need to use AmendOnDrop. All streams need to use AmendOnDrop. This was there in one commit but reverted in the latest.

LGTM if we fix vendor_extensions and AmendOnDrop

**Review by howardjohn — APPROVED** [review]
